misk-zookeeper/src/main/kotlin/misk/clustering/zookeeper/ZkLeaseCommonModule.kt

```diff
@@ -0,0 +1,48 @@
+package misk.clustering.zookeeper
+
+import com.google.common.util.concurrent.Service
+import com.google.inject.Key
+import com.google.inject.Provides
+import misk.clustering.lease.LeaseManager
+import misk.concurrent.ExecutorServiceModule
+import misk.inject.KAbstractModule
+import misk.inject.asSingleton
+import misk.inject.toKey
+import misk.tasks.RepeatedTaskQueue
+import misk.zookeeper.ZookeeperCommonModule
+import java.time.Clock
+import java.util.concurrent.ExecutorService
+import javax.inject.Singleton
+
+
+class ZkLeaseCommonModule(private val config: ZookeeperConfig) : KAbstractModule() {
+  override fun configure() {
+    install(ZkLeaseManagerModule())
+    install(ExecutorServiceModule.withFixedThreadPool(ForZkLease::class, "zk-lease-poller", 1))
+    install(ZookeeperCommonModule(config, ForZkLease::class))
+  }
+
+  companion object {
+    /** @property Key<*> the Key of the lease manager service */
+    val leaseManagerKey: Key<*> = Key.get(ZkLeaseManager::class.java)
+  }
+
+  @Provides @ForZkLease @Singleton
+  fun provideTaskQueue(
+    clock: Clock,
+    @ForZkLease executorService: ExecutorService
+  ): RepeatedTaskQueue {
+    return RepeatedTaskQueue("zk-lease-poller", clock, executorService)
+  }
+}
+
+/**
+ * Common bindings between [ZkLeaseCommonModule] and [ZkLeaseTestModule].
+ */
+internal class ZkLeaseManagerModule : KAbstractModule() {
```

Comment:
Not in love with this name, but `ZkLeaseCommonModule` was a much better name for the outer module given that it will be used in Skim. This was the best I could think of :-/